src/Lucene.Net/Search/CachingWrapperFilter.cs

```diff
@@ -79,68 +81,65 @@ public FilterCache(DeletesMode deletesMode)
 
             public T Get(IndexReader reader, object coreKey, object delCoreKey)
             {
-                lock (this)
-                {
-                    T value;
+                T value;
 
-                    if (cache == null)
-                    {
-                        cache = new WeakDictionary<object, T>();
-                    }
+                var current = _cache;
+                if (current == null)
+                {
+                    current = new ConditionalWeakTable<object, T>();
+                    current = Interlocked.CompareExchange(ref _cache, current, null) ?? current;
```

Comment:
Why do this? Any reason not to intialize this once at the ctor creation?

Reply:
Not really.. Tried to retain the original logic.

Reply:
In this case, I think we are better of doing _more_ changes :-)